src/MSBuildLocator/DotNetSdkLocationHelper.cs

```diff
@@ -15,11 +15,10 @@ internal static class DotNetSdkLocationHelper
     {
         private static readonly Regex DotNetBasePathRegex = new Regex("Base Path:(.*)$", RegexOptions.Multiline);
         private static readonly Regex VersionRegex = new Regex(@"^(\d+)\.(\d+)\.(\d+)", RegexOptions.Multiline);
+        private static readonly Regex SdkRegex = new Regex(@"(\d+\.\d+\.\d+) \[(.*)]$", RegexOptions.Multiline);
```

Comment:
This won't work for rc's, previews, and other releases with non-numeric version strings. For example:

```
.NET SDKs installed:
5.0.100-preview.8.20373.19 [C:\Program Files\dotnet\sdk]
```

The regex should be made more permissive.

Reply:
Any suggestion for a change to the regex to make it more permissive? RegEx is one of my achilles' heels :-p

Reply:
I'd try ```(\S+) \[(.*)]$```, i.e. expecting the version to be a non-empty string of non-whitespace characters.

Reply:
I tried that with "14.3.155-preview-02334alpha", and it didn't seem to work. I think \S is whitespace rather than non-whitespace characters (fixable with `^`?) and it also makes it harder to parse out groups.

I used @"(\d+)\.(\d+)\.(\d+)", and that seemed to work pretty well.

Reply:
Maybe it appears to work because it's not matching the entire version. Based on this fiddle `(\d+)\.(\d+)\.(\d+)` is not working and `\S+' looks OK: https://dotnetfiddle.net/QS5kui

Reply:
Interesting...I was testing with a local ConsoleApp that didn't agree. Did I do something wrong?
![image](https://user-images.githubusercontent.com/12969783/101798812-622a4000-3ac0-11eb-9766-bf9b74a367bb.png)

I tweaked my version to make it work: https://dotnetfiddle.net/0IUKbB

Reply:
In your screenshot `toMatch` is just a version, it does not have the space and square brackets as required by the regex.

Reply:
Ah, gotcha. Works now. 👍

Reply:
I have commited this change now.